api/src/wfl/module/covid.clj

```diff
@@ -409,16 +409,40 @@
                      :updated               now})]
       (jdbc/insert-multi! tx details (map to-rows workflows)))))
 
+(defn ^:private update-terra-workflow-statuses!
+  "Update statuses in DETAILS table for active or failed WORKSPACE workflows."
+  [{:keys [workspace details] :as _executor}]
+  (let [query  "SELECT *
+                FROM %s
+                WHERE rawls_submission_id IS NOT NULL
+                AND workflow_id IS NOT NULL
+                AND workflow_status NOT IN ('Succeeded', 'Aborted')"
```

Comment:
I felt a bit funny hard-coding this check but it's probably fine for now (worry about future-proofing though).